### DISCUSSION TITLE
Which mode do you recommend for Kupo?

### DISCUSSION THREAD

**Discussion by mrosberghaus** [geniusyield/smart-order-router#5] [Q&A] [answered]
Kupo can be operated in two modes. One is with a database with a `workdir` or `in-memory`.

What do you recommend to operate a SOR?

**Comment by sourabhxyz** [comment, marked as answer]
You should use `workdir` so that tables are persistent, o/w if you were to restart kupo, it would have to sync again.

So you can setup kupo like this:

```
kupo \
  --node-socket some-folder/cardano-node/node.socket \
  --node-config some-folder/cardano-node/config.json \
  --since origin \
  --match "*" \
  --workdir /path/to/desired/dir
```
